ibis/backends/base_sql/__init__.py

```diff
@@ -5,10 +5,21 @@
 from the SQL-based backends.
 """
 import datetime
+import itertools
 import math
+from io import StringIO
```

Comment:
pls remove from the inheriters of this (e.g. impala), otherwise you maybe importing things you don't need and/or missing things.

May want to move a very minimal amount, until we move move backends to use base_sql.

Reply:
Yup, will be doing that. I have only done the initial move from impala to base_sql so far, havent gone back to cleanup the inheriters yet.

Will there be separate issues/PRs for moving the other backends from impala to base_sql?  So ill be leaving most of impala unchanged in this PR?

Reply:
No please, I think that was some misunderstanding. Let's not make a copy of the code, move it here, and change the imports everywhere. Thanks!